src/lib/router.js

```diff
@@ -52,10 +52,23 @@ async function swapContent(url) {
 
 router
   .on("/", async () => {
-    return swapContent("index.html");
+    return swapContent("/");
   })
   .on("/*", async (params) => {
-    return swapContent(params.wild);
+    if (params.wild.endsWith("/index.html")) {
+      // If an internal link refers to "/foo/index.html", strip "index.html" and load.
+      const stripped = params.wild.slice(0, -"index.html".length);
+      return swapContent(`/${stripped}`);
+    } else if (window.location.pathname.endsWith("/")) {
+      // Navaid strips a trailing "/" on its own, so ensure it is added again before loading.
+      return swapContent(`/${params.wild}/`);
+    }
+
+    // Otherwise, this is a request for e.g. "/measure". By calling history.replaceState() to
+    // re-add the trailing slash, Navaid will re-run this code and trigger the latter conditional
+    // above. (If we also call swapContent(), two requests will fire.)
```

Comment:
> (If we also call swapContent(), two requests will fire.)

I wasn't sure what this comment was getting at. Is it saying if we call swap content here, and then update the path it'll trigger the conditional and call swapContent again? Might omit this part since it's implied. I think the comment  may have made it seem more complex than it is.

Reply:
I've made the comment much simpler.